utils/test/src/main/kotlin/OrtResultDsl.kt

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2022 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.utils.test
+
+import java.util.SortedSet
+
+import org.ossreviewtoolkit.model.AnalyzerResult
+import org.ossreviewtoolkit.model.AnalyzerRun
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageReference
+import org.ossreviewtoolkit.model.Project
+import org.ossreviewtoolkit.model.Repository
+import org.ossreviewtoolkit.model.Scope
+import org.ossreviewtoolkit.model.config.AnalyzerConfiguration
+import org.ossreviewtoolkit.utils.core.Environment
+import org.ossreviewtoolkit.utils.spdx.toSpdx
+
+@DslMarker
+annotation class OrtResultDsl
+
+@OrtResultDsl
+class OrtResultBuilder {
+    private val rootIds = mutableListOf<String>()
+    private val parentChildIds = mutableMapOf<String, MutableList<String>>()
+
+    private val packages = mutableListOf<Package>()
+
+    @OrtResultDsl
+    inner class PkgBuilder(private val id: String) {
+        @OrtResultDsl
+        var license = "Apache-2.0"
+
+        @OrtResultDsl
+        fun pkg(id: String, setup: PkgBuilder.() -> Unit): Package {
+            this@OrtResultBuilder.parentChildIds.getOrPut(this@PkgBuilder.id) { mutableListOf() } += id
```

Comment:
Why not put `parentChildIds` as constructor parameter of `PkgBuilder` to avoid doing such property access ?
You could then make `PkgBuilder` a private class in this file (not an inner class).

This has benefits that `OrtResultBuilder` can be garbage collected even if `PkgBuilder` instances are still in the memory (here its not so important but it's best practice to avoid reference to the enclosing class).